extension/lib/magento/customer/getItems.js

```diff
@@ -21,7 +21,12 @@ module.exports = async (context, input) => {
 
   const wishlistItemsEndpointUrl = `${context.config.magentoUrl}/wishlists/${wishlists[0].wishlist_id}/items`
   const wishlistItems = await request.send(wishlistItemsEndpointUrl, 'Request to Magento: getFavorites', 'GET')
-  const productIds = wishlistItems.map(item => item.product_id)
+  const wishlistItemIdMapping = {}
+  const productIds = wishlistItems.map(item => {
+    wishlistItemIdMapping[item.product_id] = item.wishlist_item_id
+    return item.product_id;
+  })
+  context.storage.user.set('wishlistItemIdMapping', wishlistItemIdMapping)
```

Comment:
u will have to kung-pow with @awesselburg, this needs to be in the setter, not the getter.

Reply:
I agree, that it has to be in the add pipeline too, but it still has to be here additionally

Reply:
The way I understood this, the merge section only adds the guest favorites to the user favorites on login/registration.
The logic here is used to get ids of favorites, that were added in the desktop shop.

Reply:
I can see a possibility where you are correct. I will think about it a bit more once we merge alexander's stuff in. Just leave it for now.